usaspending_api/etl/transaction_loaders/tests/test_fpds_loader.py

```diff
@@ -0,0 +1,269 @@
+import usaspending_api.etl.transaction_loaders.fpds_loader as fpds_loader
+import random
+from usaspending_api.etl.transaction_loaders.fpds_loader import (
+    _extract_broker_objects,
+    _create_load_object,
+    _transform_objects,
+    _load_transactions,
+)
+from usaspending_api.etl.transaction_loaders.field_mappings_fpds import (
+    transaction_fpds_columns,
+    transaction_normalized_columns,
+    legal_entity_columns,
+    legal_entity_boolean_columns,
+    recipient_location_columns,
+    recipient_location_functions,
+    place_of_performance_columns,
+    place_of_performance_functions,
+    transaction_fpds_boolean_columns,
+)
+from usaspending_api.etl.transaction_loaders.data_load_helpers import format_insert_or_update_column_sql
+
+from unittest.mock import MagicMock, patch
+
+
+def mock_cursor(monkeypatch, result_value):
+    """
+    A mock connection/cursor that ignores calls to cursor.execute and returns from fetchall
+    whatever you tell it based on mock_cursor.expected_value.  This is a bare minimum
+    implementation.  If you use any other connection or cursor features you may need to tweak
+    this accordingly (or you may not... MagicMock is, um, "magic").
+    """
+
+    class Result:
+        pass
+
+    result = Result()
+    result.expected_value = result_value
+
+    def mock_cursor():
+        mock_context = MagicMock()
+        mock_context.execute.return_value = None
+        mock_context.mogrify = mogrify
+        mock_context.fetchall.return_value = result.expected_value
+        return mock_context
+
+    def mock_connection(cursor_factory=None):
+        mock_context = MagicMock()
+        mock_context.__enter__.return_value = mock_cursor()
+        mock_context.__exit__.return_value = False
+        return mock_context
+
+    def mock_connect(dsn):
+        mock_context = MagicMock()
+        mock_context.__enter__.return_value.cursor = mock_connection
+        mock_context.__exit__.return_value = False
+        return mock_context
+
+    def mogrify(val1, val2):
+        return str(val2[0]).encode()
+
+    # I don't think it works this way...
+    result.mogrify = mogrify
+
+    monkeypatch.setattr("psycopg2.connect", mock_connect)
+
+    return result
+
+
+def _assemble_dummy_broker_data():
+    return {
+        **transaction_fpds_columns,
+        **transaction_normalized_columns,
+        **legal_entity_columns,
+        **{key: str(random.choice([True, False])) for key in legal_entity_boolean_columns},
+        **recipient_location_columns,
+        **place_of_performance_columns,
+        **{key: str(random.choice([True, False])) for key in transaction_fpds_boolean_columns},
+    }
+
+
+def _stub___extract_broker_objects(id_list):
+    """Return a list of dictionaries, in the same structure as `fetchall` when using psycopg2.extras.DictCursor"""
+    for i in range(len(id_list)):
```

Comment:
It doesn't look like range(len(...)) is necessary here. `for id in id_list:`

Reply:
this is an enumeration, not an iteration. If test style points really matter to you, we could use an enumerate function call here

Reply:
It isn't just about style, the `i` variable is _only_ used for accessing `id_list` twice. (lines 85, 86). I don't see the enumeration occurring in this function.

```python
    for item in id_list:
        dummy_row = _assemble_dummy_broker_data()
        dummy_row["detached_award_procurement_id"] = item
        dummy_row["detached_award_proc_unique"] = str(item)
        yield dummy_row
```